src/js/facility-locator/components/Select.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import Listbox from './Listbox';
+import {
+  getDirection,
+  getSelection,
+  isSelect,
+  isToggle,
+  isTraverse,
+  noServices
+} from '../utils/helpers.js';
+
+class Select extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      dropdownActive: false,
+      focusedIndex: 0
+    };
+    this.options = [];
+  }
+
+  handleFilterSelect = (newOption, type) => {
+    const { currentQuery } = this.props;
+    const { facilityType, serviceType } = currentQuery; // eslint-disable-line no-unused-vars
+    if (type === 'facility' && newOption !== facilityType) {
+      if (newOption === 'AllFacilities') {
+        this.props.updateSearchQuery({
+          facilityType: null,
+          serviceType: null
+        });
+      } else {
+        this.props.updateSearchQuery({
+          facilityType: newOption,
+          serviceType: null
+        });
+      }
+    }
+    if (type === 'service' && newOption !== serviceType) {
+      if (newOption === 'All') {
+        this.props.updateSearchQuery({
+          serviceType: null,
+        });
+      } else {
+        this.props.updateSearchQuery({
+          serviceType: newOption
+        });
+      }
+    }
+    return this.toggleDropdown(null, type);
+  }
+
+  navigateDropdown = (e) => {
+    const type = this.props.optionType;
+    const isActive = this.state.dropdownActive;
+    const which = e.target;
+    if (isToggle(e, isActive)) {
+      if (isActive && isSelect(e.keyCode)) {
+        this.handleFilterSelect(which.id, type);
+      }
+      return this.toggleDropdown(e, type);
+    }
+    if (isActive && isTraverse(e)) {
+      const increment = getDirection(e.keyCode);
+      const newIndex = this.state.focusedIndex + increment;
+      return this.focusOption(this.options[newIndex], newIndex, type);
+    }
+    return false;
+  }
+
+  handleBlur = (e) => {
+    const { currentTarget } = e;
+    setTimeout(() => {
```

Comment:
The reason for this method isn't super clear to me. Why do we need to check the activeElement?

Reply:
This ensures that the select (or one of its descendents) is currently receiving focus, otherwise (such as if the user clicks elsewhere), it will close the dropdown menu.

Reply:
It sounds like additional documentation would be useful here or perhaps some comments?

Reply:
Is that not always going to be the case if `onBlur` is triggered? It looks like this is called on the parent element of this component.

Reply:
It appears that onBlur is triggered each time a descendent is blurred (navigating through options), this way, it only toggles when the new target is not a descendent. Also, onBlur isn't triggering on Listbox, this may have something to do with how React handles reconciliation. I will create a tech debt ticket to investigate this further.